src/lazy_execution/variables.rs

```diff
@@ -0,0 +1,166 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2021, tree-sitter authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+//! Defines variable context for lazy DSL evaluation
+
+use log::debug;
+
+use crate::graph::Graph;
+use crate::Context;
+use crate::DisplayWithContext;
+use crate::Identifier;
+
+use super::store::DebugInfo;
+use super::store::Store;
+use super::values::Value;
+use super::DisplayWithContextAndGraph;
+
+/// An environment of named lazy variables
+pub trait Variables {
```

Comment:
This, for instance, _wouldn't_ need to be renamed to `LazyVariables`, because I anticipate that the other non-lazy `Variables` trait would go away when we remove the old evaluation strategy.

Reply:
Indeed. I've already merged them for another PR I'm working on, actually.